src/Microsoft.AspNetCore.Authentication/AuthenticationBuilder.cs

```diff
@@ -97,6 +97,10 @@ public EnsureSignInScheme(IOptions<AuthenticationOptions> authOptions)
             public void PostConfigure(string name, TOptions options)
             {
                 options.SignInScheme = options.SignInScheme ?? _authOptions.DefaultSignInScheme;
```

Comment:
There should be an additional  `?? _authOptions.DefaultScheme;` for when DefaultSignInScheme isn't set. That's what was starting some of the stack overlfows.

Reply:
Oops this is actually another bug, this should actually be using IAuthenticationSchemeProvider.DefaultSignInScheme I'll fix this now too

Reply:
The DefaultScheme bit needs to get backported, that's an otherwise valid configuration that currently fails.

Reply:
Fair enough so the real important fix out of all of this is for the PostConfigure to switch to using the SchemeProvider

Reply:
Ugh options being sync only bites again, so we can't cleanly use the provider since that's async, I guess we can just do the ghetto fallback using only the options